### PR TITLE
[Update 8] - Add breaking changes details for u8 release note

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
Add breaking changes related to,
1. https://github.com/ballerina-platform/ballerina-lang/issues/37750
2. https://github.com/ballerina-platform/ballerina-lang/issues/38270
3. https://github.com/ballerina-platform/ballerina-lang/issues/38498
4. https://github.com/ballerina-platform/ballerina-lang/issues/38499
5. https://github.com/ballerina-platform/ballerina-lang/pull/38029

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 1]
```diff
@@ -30,6 +30,12 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
 
 ## Backward-incompatible changes
 
+- A bug that resulted in not giving compilation errors for literal values, due to invalidly infered literal values types has been fixed.
```
Comment (SasinduDilshara) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:33]:
```suggestion
- A bug that resulted in not giving compilation errors for literal values due to invalidly inferred literal value types has been fixed.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
 
 ## Backward-incompatible changes
 
+- A bug that resulted in not giving compilation errors for literal values due to invalidly inferred literal value types has been fixed.
```
Comment (pcnfernando) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:33]:
```suggestion
A bug related to the inference of numeric literal types has been fixed. 
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 3]
```diff
@@ -30,6 +30,12 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
...
+
+    ```ballerina
+    2f|1 num = 2; // Compilation error now.
```
Comment (pcnfernando) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:36]: @SasinduDilshara @LakshanWeerasinghe, can we add a few more applicable scenarios here.

**Comment by pcnfernando** [comment]
@SasinduDilshara @LakshanWeerasinghe point out the issues that this specific release note section addresses in the PR description

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md (hunk 3 above)
Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:36]: @LakshanWeerasinghe few examples are

```
2f|3d|6 num = 10; // Compilation error now.
3d|6 num = 10; // Compilation error now.
```

Comment (LakshanWeerasinghe) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md]: Updated.

**Comment by LakshanWeerasinghe** [comment]
> @SasinduDilshara @LakshanWeerasinghe point out the issues that this specific release note section addresses in the PR description

updated.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 4]
```diff
@@ -30,6 +30,14 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
 
 ## Backward-incompatible changes
 
+- A bug related to the inference of numeric literal types has been fixed.
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:33]: Please see https://github.com/ballerina-platform/ballerina-dev-website/pull/7642#discussion_r1305092375.

**Review by pcnfernando — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 5]
```diff
@@ -32,6 +32,14 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
 
 - Ballerina interoperability implementation may have an impact with the Java 17 support due to any incompatible changes. For example, Java 17 has some restrictions on using Java reflections with internal Java packages. For more information, see the Java 17 release notes.
 
+- A bug related to the inference of numeric literal types has been fixed.
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:35]:
```suggestion
- A bug related to deciding the types of numeric literals has been fixed.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 6]
```diff
@@ -32,6 +32,14 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
...
+
+    ```ballerina
+    2f|1 num1 = 2; // Compilation error now.
+    2f|3d|6 num2 = 10; // Compilation error now.
+    3d|6 num3 = 10; // Compilation error now.
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:40]:
```suggestion
    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.
    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
```

Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:40]: These are already errors, not sure what the second one (L39) meant to show.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 7]
```diff
@@ -32,6 +32,13 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
 
 - Ballerina interoperability implementation may have an impact with the Java 17 support due to any incompatible changes. For example, Java 17 has some restrictions on using Java reflections with internal Java packages. For more information, see the Java 17 release notes.
 
+- A bug related to deciding the types of numeric literals has been fixed.
+
+    ```ballerina
+    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.
+    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
```
Comment (SasinduDilshara) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:39]:
```suggestion
    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
    3d|4f|6 b = 4; // Compilation error now, `` is considered to be `int`.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md [hunk 8]
```diff
@@ -32,6 +32,14 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
...
+    3d|4f|6 b = 4; // Compilation error now, `4` is considered to be `int`.
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md:40]: Similar to the first one? It's OK though.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**